src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -118,19 +116,23 @@ public boolean Disconnect(String peerId){
     public boolean  DisconnectIncomingConnections(){
 
         boolean ret = false;
+
         if (mServerSocketList != null) {
-            for(int i = 0; i < mServerSocketList.size(); i++) {
-                BtToServerSocket tmp = mServerSocketList.get(i);
-                if (tmp != null) {
-                    print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
-                    tmp.Stop();
-                    tmp = null;
-                    mServerSocketList.set(i, tmp);
-                    ret = true;
+            synchronized(mServerSocketList) {
+
+                for (int i = 0; i < mServerSocketList.size(); i++) {
+                    BtToServerSocket tmp = mServerSocketList.get(i);
+                    if (tmp != null) {
+                        print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
+                        tmp.Stop();
+                        tmp = null;
+                        mServerSocketList.set(i, tmp);
+                        ret = true;
```

Comment:
I don't understand the intended semantics of 'ret'. It seems to mean "I found and closed at least one server socket". But why is that useful? I would have thought that you would only want to return true if you successfully closed all the sockets? And if there were no sockets then shouldn't you still return true since you did close all sockets, there just weren't any?